arithmetization/src/main/java/net/consensys/linea/extradata/LineaExtraDataHandler.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.extradata;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.config.LineaProfitabilityConfiguration;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt32;
+import org.hyperledger.besu.plugin.data.AddedBlockContext;
+import org.hyperledger.besu.plugin.services.BesuEvents;
+
+@Slf4j
+public class LineaExtraDataHandler implements BesuEvents.BlockAddedListener {
+  private final ExtraDataParser[] extraDataParsers;
+
+  public LineaExtraDataHandler(final LineaProfitabilityConfiguration profitabilityConf) {
+    extraDataParsers = new ExtraDataParser[] {new Version1Parser(profitabilityConf)};
+  }
+
+  @Override
+  public void onBlockAdded(final AddedBlockContext addedBlockContext) {
+    final var blockHeader = addedBlockContext.getBlockHeader();
+    final var rawExtraData = blockHeader.getExtraData();
+
+    if (!Bytes.EMPTY.equals(rawExtraData)) {
+      for (final ExtraDataParser extraDataParser : extraDataParsers) {
+        if (extraDataParser.canParse(rawExtraData)) {
+          final var extraData = rawExtraData.slice(1);
+          extraDataParser.parse(extraData);
+          return;
+        }
+      }
+      log.warn("unsupported extra data field {}", rawExtraData.toHexString());
+    }
+  }
+
+  private interface ExtraDataParser {
+    boolean canParse(Bytes extraData);
+
+    void parse(Bytes extraData);
+
+    static Long toLong(final Bytes fieldBytes) {
+      return UInt32.fromBytes(fieldBytes).toLong();
+    }
+  }
+
+  @SuppressWarnings("rawtypes")
+  private static class Version1Parser implements ExtraDataParser {
```

Comment:
Just cross-posting Coordinator implementation of codec for history and for easier comparison https://github.com/Consensys/zkevm-monorepo/pull/3186/files#diff-ae13fbcae1669b37d87d5cef4caf8751d825779836d7932946021096564697f7R22